src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -36,11 +36,20 @@ public DeleteCommand(Index targetIndex) {
 
     @Override
     public CommandResult executeUndoableCommand() {
-        requireNonNull(personToDelete);
-        try {
-            model.deletePerson(personToDelete);
-        } catch (PersonNotFoundException pnfe) {
-            throw new AssertionError("The target person cannot be missing");
+        if (personToDelete != null && studentToDelete == null) {
+            try {
+                model.deletePerson(personToDelete);
+            } catch (PersonNotFoundException pnfe) {
+                throw new AssertionError("The target person cannot be missing");
```

Comment:
I see  "The target person cannot be missing" at some place and "The target contact cannot be missing" at some other. Maybe you can make a constant variable to ensure consistency.